pyomo/contrib/alternative_solutions/solution.py

```diff
@@ -9,150 +9,285 @@
 #  This software is distributed under the 3-clause BSD License.
 #  ___________________________________________________________________________
 
+import sys
+import heapq
+import collections
+import dataclasses
 import json
+import functools
+
 import pyomo.environ as pyo
-from pyomo.common.collections import ComponentMap, ComponentSet
-from pyomo.contrib.alternative_solutions import aos_utils
 
+from .aos_utils import MyMunch, _to_dict
 
-class Solution:
+nan = float("nan")
+
+
+def _custom_dict_factory(data):
+    return {k: _to_dict(v) for k, v in data}
+
+
+if sys.version_info >= (3, 10):
+    dataclass_kwargs = dict(kw_only=True)
+else:
+    dataclass_kwargs = dict()
+
+
+@dataclasses.dataclass(**dataclass_kwargs)
+class VariableInfo:
     """
-    A class to store solutions from a Pyomo model.
+    Represents a variable in a solution.
 
     Attributes
     ----------
-    variables : ComponentMap
-        A map between Pyomo variables and their values for a solution.
-    fixed_vars : ComponentSet
-        The set of Pyomo variables that are fixed in a solution.
-    objective : ComponentMap
-        A map between Pyomo objectives and their values for a solution.
-
-    Methods
-    -------
-    pprint():
-        Prints a solution.
-    get_variable_name_values(self, ignore_fixed_vars=False):
-        Get a dictionary of variable name-variable value pairs.
-    get_fixed_variable_names(self):
-        Get a list of fixed-variable names.
-    get_objective_name_values(self):
-        Get a dictionary of objective name-objective value pairs.
+    value : float
+        The value of the variable.
+    fixed : bool
+        If True, then the variable was fixed during optimization.
+    name : str
+        The name of the variable.
+    index : int
+        The unique identifier for this variable.
+    discrete : bool
+        If True, then this is a discrete variable
+    suffix : dict
+        Other information about this variable.
     """
 
-    def __init__(self, model, variable_list, include_fixed=True, objective=None):
-        """
-        Constructs a Pyomo Solution object.
+    value: float = nan
+    fixed: bool = False
+    name: str = None
+    repn = None
+    index: int = None
+    discrete: bool = False
+    suffix: MyMunch = dataclasses.field(default_factory=MyMunch)
+
+    def to_dict(self):
+        return dataclasses.asdict(self, dict_factory=_custom_dict_factory)
+
+
+@dataclasses.dataclass(**dataclass_kwargs)
+class ObjectiveInfo:
+    """
+    Represents an objective in a solution.
+
+    Attributes
+    ----------
+    value : float
+        The objective value.
+    name : str
+        The name of the objective.
+    index : int
+        The unique identifier for this objective.
+    suffix : dict
+        Other information about this objective.
+    """
```

Comment:
See my comment above, but wondering if storing a reference to the original Objective might not be a good thing here--that way someone could get the expression or the sense or whatever they wanted.

Reply:
See my note above that this would require a separate PR.  The Solution object, and its constituent components, was designed to be agnostic of Pyomo.